### PR TITLE
Initial 'id as Any' Clang importer implementation

### PR DESCRIPTION
**Pull request by jckarter**
Adds a frontend flag `-enable-id-as-any` to gate importing ObjC `id` as the Swift `Any` type. Initial Clang importer and representable-in-ObjC support.

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -1500,6 +1500,12 @@ namespace {
 
       Type SwiftType;
       if (Decl->getDeclContext()->getRedeclContext()->isTranslationUnit()) {
+        // Ignore the 'id' typedef. We want to bridge the underlying
+        // ObjCId type.
+        if (Name.str() == "id" && Impl.SwiftContext.LangOpts.EnableIdAsAny) {
```
Comment (jrose-apple) [inline lib/ClangImporter/ImportDecl.cpp]: We already don't import the 'id' typedef.

Comment (jckarter) [inline lib/ClangImporter/ImportDecl.cpp]: From what I've observed, we *do* recognize it from the table in `BridgedTypes.def` and immediately map it to `AnyObject` without this early exit.

Comment (jrose-apple) [inline lib/ClangImporter/ImportDecl.cpp]: Ah, and we can't make BridgedTypes.def conditional. Okay.

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci  Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
Failure looks unrelated:

```swift
******************** TEST 'Swift :: stdlib/String.swift' FAILED ********************
Script:
--
rm -rf /Users/buildnode/jenkins/workspace/swift-PR-osx/buildbot_incremental/swift-macosx-x86_64/validation-test-macosx-x86_64/stdlib/Output/String.swift.tmp && mkdir -p /Users/buildnode/jenkins/workspace/swift-PR-osx/buildbot_incremental/swift-macosx-x86_64/validation-test-macosx-x86_64/stdlib/Output/String.swift.tmp && xcrun --toolchain default --sdk /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk /Users/buildnode/jenkins/workspace/swift-PR-osx/buildbot_incremental/swift-macosx-x86_64/bin/swiftc -target x86_64-apple-macosx10.9  -module-cache-path '/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/swift-testsuite-clang-module-cachelcOcKP' -F /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/../../../Developer/Library/Frameworks -Xlinker -rpath -Xlinker /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/../../../Developer/Library/Frameworks    -module-cache-path '/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/swift-testsuite-clang-module-cachelcOcKP' /Users/buildnode/jenkins/workspace/swift-PR-osx/swift/validation-test/stdlib/String.swift -o /Users/buildnode/jenkins/workspace/swift-PR-osx/buildbot_incremental/swift-macosx-x86_64/validation-test-macosx-x86_64/stdlib/Output/String.swift.tmp/a.out -module-name main &&  /Users/buildnode/jenkins/workspace/swift-PR-osx/buildbot_incremental/swift-macosx-x86_64/validation-test-macosx-x86_64/stdlib/Output/String.swift.tmp/a.out
--
Exit Code: 1

Command Output (stderr):
--
/Users/buildnode/jenkins/workspace/swift-PR-osx/swift/validation-test/stdlib/String.swift:205:11: warning: expression of type 'Character' is unused
  acceptor[i]
  ~~~~~~~~^~~
/Users/buildnode/jenkins/workspace/swift-PR-osx/swift/validation-test/stdlib/String.swift:232:11: warning: expression of type 'String' is unused
  acceptor[r]
  ~~~~~~~~^~~
/Users/buildnode/jenkins/workspace/swift-PR-osx/swift/validation-test/stdlib/String.swift:243:11: warning: expression of type 'String' is unused
  acceptor[r]
  ~~~~~~~~^~~
/Users/buildnode/jenkins/workspace/swift-PR-osx/swift/validation-test/stdlib/String.swift:1160:26: error: cannot call value of non-function type 'NSRange' (aka '_NSRange')
    let r = result!.range(at: 1)
            ~~~~~~~~~~~~~^
```

**Pull request merged**